repos/spack_repo/builtin/packages/ripgrep/package.py

```diff
@@ -27,22 +27,37 @@ class Ripgrep(CargoPackage):
     version("13.0.0", sha256="0fb17aaf285b3eee8ddab17b833af1e190d73de317ff9648751ab0660d763ed2")
     version("11.0.2", sha256="0983861279936ada8bc7a6d5d663d590ad34eb44a44c75c2d6ccd0ab33490055")
 
+    variant("pcre2", default=False, description="Add support for Perl-style regex via PCRE2")
+
     depends_on("rust@1.85:", type="build", when="@15:")
-    depends_on("rust@1.72:", type="build", when="@14:")
+    depends_on("rust@1.72:", type="build", when="@14")
+    depends_on("rust@1.31:", type="build", when="@:13")
```

Comment:
```suggestion
    depends_on("rust@1.72:", type="build", when="@14:")
    depends_on("rust@1.31:", type="build")
```

I would drop the `when="@:13"` from here as you're effectively creating a base case for all older versions. I'd then retain the `when="@14:"` (colon at the end) such that the newer major version's requirements layer on top of each other.

Reply:
Addressed with 9645980.